catalyst_voices/packages/internal/catalyst_voices_models/lib/src/proposal_schema/proposal_definitions.dart

```diff
@@ -0,0 +1,126 @@
+enum DefinitionsObjectTypes {
+  string,
+  object,
+  integer,
+  array,
+  unknown;
+
+  const DefinitionsObjectTypes();
+
+  static DefinitionsObjectTypes fromString(String value) {
+    return DefinitionsObjectTypes.values.firstWhere(
+      (type) => type.name == value,
+      orElse: () => DefinitionsObjectTypes.unknown,
+    );
+  }
+}
+
+enum DefinitionsContentMediaType {
+  textPlain('text/plain'),
+  markdown('text/markdown'),
+  unknown('unknown');
+
+  final String value;
+
+  const DefinitionsContentMediaType(this.value);
+
+  static DefinitionsContentMediaType fromString(String value) {
+    return DefinitionsContentMediaType.values.firstWhere(
+      (type) => type.name == value,
+      orElse: () => DefinitionsContentMediaType.unknown,
+    );
+  }
+}
+
+enum DefinitionsFormats {
+  singleLineTextEntry,
+  multiLineTextEntry,
+  multiLineTextEntryMarkdown,
+  dropDownSingleSelect,
+  multiSelect,
+  singleLineHttpsURLEntry,
+  nestedQuestions,
+  nestedQuestionsList,
+  singleGroupedTagSelector,
+  tagGroup,
+  tagSelection,
+  tokenValueCardanoADA,
+  durationInMonths,
+  yesNoChoice,
+  agreementConfirmation,
+  spdxLicenseOrURL,
+}
+
+abstract class BaseProposalDefinition {
+  final DefinitionsObjectTypes type;
+  final String note;
+
+  const BaseProposalDefinition({
+    required this.type,
+    required this.note,
+  });
+}
+
+class ProposalDefinitions {
+  final SegmentProposalDefinition segmentDefinition;
+  final SectionProposalDefinition sectionDefinition;
+  final SingleLineTextEntryDefinition singleLineTextEntryDefinition;
```

Comment:
`SingleLineTextEntryDefinition singleLineTextEntryDefinition` it seems strange. What is purpose of this property and class in general?